packages/replace/src/index.js

```diff
@@ -37,14 +37,22 @@ function mapToFunctions(object) {
 
 export default function replace(options = {}) {
   const filter = createFilter(options.include, options.exclude);
-  const { delimiters } = options;
+  const { delimiters, preventAssignment } = options;
   const functionValues = mapToFunctions(getReplacements(options));
-  const keys = Object.keys(functionValues)
-    .sort(longest)
-    .map(escape);
+  const keys = Object.keys(functionValues).sort(longest).map(escape);
+  if (![true, false].includes(preventAssignment)) {
+    // eslint-disable-next-line no-console
+    console.warn(
```

Comment:
```suggestion
    this.warn(
```

we'll use the plugin system's warning mechanism